### DISCUSSION TITLE
ReactiveComponent : the component is never removed from the store

### DISCUSSION THREAD

**Discussion by bdecarne** [appbaseio/reactivesearch#1830] [General]
**Affected Projects**

React

**Library Version:** x.y.z

3.26.1-alpha.2

**Describe the bug**

A rendered ReactiveComponent will never be removed from the store. In the commit https://github.com/appbaseio/reactivesearch/commit/8d01178c97b360124f300c3f73429cb66cd5b404 a new `destroyOnUnmount`property has been added to `ComponentWrapper`. Unfortunately, this prop is always undefined, the `removeComponent` action is not called on component unmount.

**Comment by bietkul** [comment]
@bdecarne The commit that you've shared fixes an important issue where force re-render breaks the components. I would like to understand the use case where you don't want to keep the `ReactiveComponent` in store. If `ReactiveComponent` value/query is a concern, then it can be cleared on unmount by using a wrapper component.
However, you can set the `destroyOnUnmount` as `true` to `ReactiveComponent`.

**Comment by bdecarne** [comment]
Hi, thank for your reply !

> I would like to understand the use case where you don't want to keep the ReactiveComponent in store

I use `ReactiveComponent` to wrap 2 custom listing components in the same application. I use it to display results in two distinct view modes, "list" and "grid" : components are not displayed at the same time.

Both components has dependency to a MultiList through `react` prop : each time I select an option, the active listing components is doing a new request. 

But, when I switch to the other listing component (so the first is unmount) and a choose a new option in the multilist, I see the request of the unmounted component in the _msearch call to the server. **The _msearch contains all requests for all components, even those which has been unmounted** : it's a performance leak.

> If ReactiveComponent value/query is a concern, then it can be cleared on unmount by using a wrapper component.

As i described it, unmount `ReactComponent` will not remove the component from store

> However, you can set the destroyOnUnmount as true to ReactiveComponent

I use Typescript, `destroyOnUnmount` is not a valid prop of `ReactiveComponent`

**Comment by bdecarne** [comment]
I have a lot of other situations in my application : for example, i use conditional rules from `StateProvider` to render a specific `MultiList` component when a specific value has been selected from other `MultiList`. When I deselect the option, conditionnal `MultiList` is unmounted, but the corresponding query stay in _msearch.

I would like to understand the use case where you want to keep the ReactiveComponent in store when the corresponding component is unmounted ?

For information, all my troubles went away when i patched the `ComponentWrapper` to add : 

```ts
ComponentWrapper.defaultProps = {
  destroyOnUnmount: true
};
```

I wish you a sweet christmas eve :santa:

**Comment by bdecarne** [comment]
@bietkul As I see in fb18f0d61649b43de47cc7696b17a32cd2e6d009, you finally decided to integrate this request into the basecode. Thank you.
